src/net_processing.cpp

```diff
@@ -3887,7 +3889,10 @@ bool PeerManager::ProcessMessages(CNode* pfrom, std::atomic<bool>& interruptMsgP
     // this maintains the order of responses
     // and prevents vRecvGetData to grow unbounded
     if (!pfrom->vRecvGetData.empty()) return true;
-    if (!peer->m_orphan_work_set.empty()) return true;
+    {
+        LOCK(g_cs_orphans);
```

Comment:
673247b58cd1252ab7e99f7d63ead05cc100cef2, nit:
```suggestion
        LOCK(::g_cs_orphans);
```